pkg/builder/controller.go

```diff
@@ -228,26 +238,33 @@ func (blder *Builder) loadRestConfig() {
 	}
 }
 
-func (blder *Builder) getControllerName() (string, error) {
+func (blder *Builder) getControllerName(gvk schema.GroupVersionKind) string {
 	if blder.name != "" {
-		return blder.name, nil
+		return blder.name
 	}
-	gvk, err := getGvk(blder.forInput.object, blder.mgr.GetScheme())
-	if err != nil {
-		return "", err
-	}
-	return strings.ToLower(gvk.Kind), nil
+	return strings.ToLower(gvk.Kind)
 }
 
 func (blder *Builder) doController(r reconcile.Reconciler) error {
-	name, err := blder.getControllerName()
-	if err != nil {
-		return err
-	}
 	ctrlOptions := blder.ctrlOptions
 	if ctrlOptions.Reconciler == nil {
 		ctrlOptions.Reconciler = r
 	}
-	blder.ctrl, err = newController(name, blder.mgr, ctrlOptions)
+
+	// Retrieve the GVK from the object we're reconciling
+	// to prepopulate logger information, and to optionally generate a default name.
+	gvk, err := getGvk(blder.forInput.object, blder.mgr.GetScheme())
+	if err != nil {
+		return err
+	}
+
+	// Setup the logger.
+	if ctrlOptions.Log == nil {
+		ctrlOptions.Log = logf.Log
+	}
+	ctrlOptions.Log = ctrlOptions.Log.WithValues("groupVersionKind", gvk.String())
```

Comment:
Do we want to split this into multiple values, or is `groupVersionKind` enough here?

Reply:
I think gvk is probably good here, I do kind of wish we could abbreviate it to `gvk` for the key for brevity, but it's probably better to keep it spelled out. The length is also pretty minimal compared to a full gvk anyway :joy:

Reply:
Seems gvk.String() is `return gvk.Group + "/" + gvk.Version + ", Kind=" + gvk.Kind`, that might be a little weird to parse for folks. I'm also ok to keep it this way and spell it as `gvk`, given that `groupVersionKind` is kinda large in log lines.

Is the group and version information actually useful? Or do we wanto just output the kind?

Reply:
I kind of wish we could have different behavior based on log level :joy: I think Kind is probably sufficient in most cases, but I also know that we've wanted to output version information in the past as well (mainly around upgrade scenarios).

Reply:
For `Version` I am not sure if its useful because the different versions are supposively just different views on the same data.

For `Group` and `Kind` we definitely should have both, some clusters have multiple identically named kinds in different api groups (`Cluster` is a common example for this :upside_down_face: )

Reply:
I'll split them up!